### PR TITLE
global: communities sorting options

### PR DESCRIPTION
**Pull request by krzysztof**

### CODE REVIEW

File: invenio_communities/models.py [hunk 1]
```diff
@@ -175,6 +175,12 @@ class Community(db.Model, Timestamp):
     logo_ext = db.Column(db.String(length=5), nullable=True, default=None)
     """Extension of the logo."""
 
+    ranking = db.Column(db.Integer, nullable=False, default=0)
+    """ Ranking of community. Updated by ranking deamon"""
```
Comment (jirikuncar) [inline invenio_communities/models.py]: please remove space from the beginning of docstring

File: invenio_communities/models.py [hunk 2]
```diff
@@ -175,6 +175,12 @@ class Community(db.Model, Timestamp):
...
+
+    fixed_points = db.Column(db.Integer, nullable=False, default=0)
+    """ Points which will be always added to overall score of community"""
```
Comment (jirikuncar) [inline invenio_communities/models.py]: see https://github.com/inveniosoftware/invenio-communities/pull/16/files#r57283070

File: invenio_communities/templates/invenio_communities/index.html [hunk 3]
```diff
@@ -52,8 +53,28 @@
...
+                  <a href="{{ url_for('invenio_communities.index', **new_args) }}" class="active">
+                    <i class="pull-right icon {{ ' glyphicon glyphicon-ok' if args.get('so', 'ranking')==opt }}"></i>
+                    {{ opt }}
+                  </a>
+                </li>
+              {%- endfor -%}
+              </ul>
+            </div>
+          </div>
+
```
Comment (jirikuncar) [inline invenio_communities/templates/invenio_communities/index.html]: extra empty line?

File: invenio_communities/templates/invenio_communities/portalbox_main.html [hunk 4]
```diff
@@ -30,7 +30,7 @@
 {%- if community.description %}
 {{community.description|safe}}
 {%- endif %}
-{% if community.page and not detail %}<a href="{{url_for('communities.detail', community_id=community.id)}}" class="pull-right">Read more</a><br>{% endif %}
+{% if community.page and not detail %}<a href="{{url_for('invenio_communities.detail', community_id=community.id)}}" class="pull-right">Read more</a><br>{% endif %}
```
Comment (jirikuncar) [inline invenio_communities/templates/invenio_communities/portalbox_main.html]:
```jinja2
{%- if community.page and not detail -%}
<a href="{{ url_for('invenio_communities.detail', community_id=community.id) }}" class="pull-right">
  {{ _('Read more') }}
</a><br>
{%- endif %}
```

File: invenio_communities/views.py [hunk 5]
```diff
@@ -62,6 +62,9 @@ def index(p, so, page):
     """Index page with uploader and list of existing depositions."""
     ctx = mycommunities_ctx()
 
+    if not so:
+        so = current_app.config.get('COMMUNITIES_DEFAULT_SORTING_OPTION')
```
Comment (jirikuncar) [inline invenio_communities/views.py]: `so = so or current_app.config.get('COMMUNITIES_DEFAULT_SORTING_OPTION')`

File: invenio_communities/templates/invenio_communities/index.html [hunk 6]
```diff
@@ -52,8 +53,27 @@
...
+                <i class="glyphicon glyphicon-random"></i> {{ _('Sort by') }}
+                <span class="caret"></span>
+              </a>
+              <ul class="dropdown-menu">
+              {%- for opt in ['title', 'ranking'] -%}
+                {%- set new_args = args.copy() -%}
+                {%- do new_args.update({'so': opt}) -%}
+                <li>
+                  <a href="{{ url_for('invenio_communities.index', **new_args) }}" class="active">
+                    <i class="pull-right icon {{ ' glyphicon glyphicon-ok' if args.get('so', 'ranking')==opt }}"></i>
```
Comment (jirikuncar) [inline invenio_communities/templates/invenio_communities/index.html]: `' glyphicon` -> `'glyphicon` ?

File: invenio_communities/templates/invenio_communities/portalbox_main.html [hunk 7]
```diff
@@ -28,17 +28,20 @@
...
-    <dt>Title:</dt><dd>{{community.title}}</dd>
-    {# <dt>Curated by:</dt><dd><a href="{{url_for('yourmessages.write', sent_to_user_nicks=community.owner.nickname)}}">{{community.owner.nickname}}</a></dd> #}
-    <dt>Curated by:</dt><dd>{{community.owner.nickname}}</dd>
-    <dt>Curation policy:</dt><dd>{% if community.curation_policy %}{{community.curation_policy}}{% else %}Not specified{% endif %}</dd>
-    <dt>Created:</dt><dd>{{community.created.strftime('%Y-%m-%d')}}</dd>
-    <dt>Harvesting API:</dt><dd><a href="{{community.oai_url}}">OAI-PMH Interface</a></dd>
+  <dt>{{ _('Title') }}:</dt><dd>{{ community.title }}</dd>
+  {# <dt>{{ _('Curated by') }}:</dt><dd><a href="{{ url_for('yourmessages.write', sent_to_user_nicks=community.owner.nickname) }}">{{ community.owner.nickname }}</a></dd> #}
+  <dt>{{ _('Curated by') }}:</dt><dd>{{ community.owner.nickname }}</dd>
+  <dt>{{ _('Curation policy') }}:</dt><dd>{% if community.curation_policy %}{{ community.curation_policy }}{% else %}{{ _('Not specified') }}{% endif %}</dd>
```
Comment (jirikuncar) [inline invenio_communities/templates/invenio_communities/portalbox_main.html]: `{{ community.curation_policy|default(_('Not specified')) }}` ?

File: invenio_communities/templates/invenio_communities/portalbox_main.html [hunk 8]
```diff
@@ -28,17 +28,20 @@
...
+{% endif %}
+<hr />
 <dl>
-    <dt>Title:</dt><dd>{{community.title}}</dd>
-    {# <dt>Curated by:</dt><dd><a href="{{url_for('yourmessages.write', sent_to_user_nicks=community.owner.nickname)}}">{{community.owner.nickname}}</a></dd> #}
-    <dt>Curated by:</dt><dd>{{community.owner.nickname}}</dd>
-    <dt>Curation policy:</dt><dd>{% if community.curation_policy %}{{community.curation_policy}}{% else %}Not specified{% endif %}</dd>
-    <dt>Created:</dt><dd>{{community.created.strftime('%Y-%m-%d')}}</dd>
-    <dt>Harvesting API:</dt><dd><a href="{{community.oai_url}}">OAI-PMH Interface</a></dd>
+  <dt>{{ _('Title') }}:</dt><dd>{{ community.title }}</dd>
```
Comment (jirikuncar) [inline invenio_communities/templates/invenio_communities/portalbox_main.html]: we should probably keep `:` with the translation string for RTL languages.

Comment (krzysztof) [inline invenio_communities/templates/invenio_communities/portalbox_main.html]: Colons are now in the translation string.

**Comment by jirikuncar** [comment]
:shipit:

**Pull request merged**